cognite/client/data_classes/_base.py

```diff
@@ -191,7 +191,9 @@ def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None =
         return cls(resource)
 
     def dump(self, camel_case: bool = True) -> dict[str, Any]:
-        return convert_all_keys_recursive(self.__data, camel_case=camel_case)
+        if isinstance(self.__data, dict):
```

Comment:
Not necessary, this should always be a dict. If it can be anything else, then the `_init_` and `load` signatures should be updated.